eth2near/contract_wrapper/src/dao_contract.rs

```diff
@@ -0,0 +1,216 @@
+    Approve = 0x0,
+    Reject = 0x1,
+    Remove = 0x2,
+}
+
+/// Proposal that are sent to this DAO.
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize)]
+#[cfg_attr(not(target_arch = "wasm32"), derive(Debug))]
+#[serde(crate = "near_sdk::serde")]
+pub struct Proposal {
+    /// Original proposer.
+    pub proposer: AccountId,
+    /// Description of this proposal.
+    pub description: String,
+    /// Kind of proposal with relevant information.
+    pub kind: ProposalKind,
+    /// Current status of the proposal.
+    pub status: ProposalStatus,
+    /// Count of votes per role per decision: yes / no / spam.
+    pub vote_counts: HashMap<String, [Balance; 3]>,
+    /// Map of who voted and how.
+    pub votes: HashMap<AccountId, Vote>,
+    /// Submission time (for voting period).
+    pub submission_time: U64,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(crate = "near_sdk::serde")]
+pub struct ProposalInput {
+    /// Description of this proposal.
+    pub description: String,
+    /// Kind of proposal with relevant information.
+    pub kind: ProposalKind,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Debug)]
+#[serde(crate = "near_sdk::serde")]
+pub enum Action {
+    /// Action to add proposal. Used internally.
+    AddProposal,
+    /// Action to remove given proposal. Used for immediate deletion in special cases.
+    RemoveProposal,
+    /// Vote to approve given proposal or bounty.
+    VoteApprove,
+    /// Vote to reject given proposal or bounty.
+    VoteReject,
+    /// Vote to remove given proposal or bounty (because it's spam).
+    VoteRemove,
+    /// Finalize proposal, called when it's expired to return the funds
+    /// (or in the future can be used for early proposal closure).
+    Finalize,
+    /// Move a proposal to the hub to shift into another DAO.
+    MoveToHub,
+}
+
+pub struct DAOContract {
+    last_proposal_id: u64,
+    contract_wrapper: Box<dyn ContractWrapper>,
+}
+
+impl DAOContract {
+    pub fn new(contract_wrapper: Box<dyn ContractWrapper>) -> Self {
+        DAOContract {
+            last_proposal_id: 0,
+            contract_wrapper,
+        }
+    }
+
+    pub fn get_last_added_proposal(&self) -> u64 {
+        return self.last_proposal_id;
+    }
+
+    pub fn get_last_proposal_id(&self) -> Result<u64, Box<dyn Error>> {
+        let response = self
+            .contract_wrapper
+            .call_view_function("get_last_proposal_id".to_string(), vec![])?;
+
+        Ok(serde_json::from_slice(response.as_slice())?)
+    }
+
+    pub fn get_proposal(&self, id: u64) -> Result<ProposalOutput, Box<dyn Error>> {
+        let response = self.contract_wrapper.call_view_function(
+            "get_proposal".to_string(),
+            json!({ "id": id }).to_string().into_bytes(),
+        )?;
+
+        Ok(serde_json::from_slice(response.as_slice())?)
+    }
+
+    pub fn add_proposal(&mut self, proposal: ProposalInput) -> Result<u64, Box<dyn Error>> {
```

Comment:
Is this function should be public? Or it is going to be use only inside submit_light_client_update_proposale?

Reply:
This is not an internal method, so it should be public even if only the helper method is used now.